Zend/zend_compile.c

```diff
@@ -7384,6 +7422,24 @@ void zend_compile_class_decl(znode *result, zend_ast *ast, zend_bool toplevel) /
 		zend_verify_abstract_class(ce);
 	}
 
+	if (ce->ce_flags & ZEND_ACC_ENUM) {
+		zval case_default_value;
+		ZVAL_NULL(&case_default_value);
+		zend_string *case_property_name = zend_string_init("case", strlen("case"), 0);
+		zend_type case_type = ZEND_TYPE_INIT_NONE(0);
+		zend_declare_typed_property(ce, case_property_name, &case_default_value, ZEND_ACC_PUBLIC, NULL, case_type);
+		zend_string_release(case_property_name);
+
+		if (ce->enum_primitive_type != IS_UNDEF) {
+			zval value_default_value;
+			ZVAL_NULL(&value_default_value);
+			zend_string *vaue_property_name = zend_string_init("value", strlen("value"), 0);
+			zend_type value_type = ZEND_TYPE_INIT_NONE(0);
+			zend_declare_typed_property(ce, vaue_property_name, &value_default_value, ZEND_ACC_PUBLIC, NULL, value_type);
```

Comment:
Could document what this is doing - does `ZEND_TYPE_INIT_NONE(0)` mean that it's impossible to assign any type to it?

Could also add a test such as `unset(Suit::HEARTS->value);` - will that need additional code changes to prevent?

Reply:
The type is now set to the enum type. Also added a test for unset.